client/app/bundles/course/video/attemptLoader.ts

```diff
@@ -0,0 +1,29 @@
+import { LoaderFunction, Params, redirect } from 'react-router-dom';
+import { getIdFromUnknown } from 'utilities';
+
+import CourseAPI from 'api/course';
+
+const getSubmissionURL = (
+  params: Params<string>,
+  submissionId: number,
+): string | null => {
+  const courseId = getIdFromUnknown(params?.courseId);
+  const videoId = getIdFromUnknown(params?.videoId);
+  if (!courseId || !videoId) return null;
+
+  return `/courses/${courseId}/videos/${videoId}/submissions/${submissionId}/edit`;
+};
+
+const videoAttemptLoader: LoaderFunction = async ({ params }) => {
+  const videoId = getIdFromUnknown(params?.videoId);
+  if (!videoId) return redirect('/');
+
+  const { data } = await CourseAPI.video.submissions.create(videoId);
+
+  const url = data.submissionUrl ?? getSubmissionURL(params, data.submissionId);
```

Comment:
Perhaps, the backend only needs to return the submissionId and no need for submissionUrl anymore? This way we dont need to do extra check if submissionUrl exists in the frontend